pkg/gateway/testutil/gateway_setup.go

```diff
@@ -65,7 +66,13 @@ func GetBasicHandler(t *testing.T, authService *FakeAuthService, repoName string
 	_, err = c.CreateRepository(ctx, repoName, "", storageNamespace, "main", false)
 	testutil.Must(t, err)
 
-	handler := gateway.NewHandler(authService.Region, c, multipartTracker, blockAdapter, authService, []string{authService.BareDomain}, &stats.NullCollector{}, upload.DefaultPathProvider, nil, config.DefaultLoggingAuditLogLevel, true, false, false)
+	logger := logging.Dummy()
+	middlewareFactory, err := gatewayfactory.BuildMiddleware(ctx, cfg, logger)
+	if err != nil {
+		logger.WithError(err).Fatal("Failed to create gateway middleware")
+	}
```

Comment:
I don't understand this code.  Why drop all the messages in an integration test?  For instance, right here on l.72 the code will abort the program with no error message whatsoever.
Why no use a real logger, which will show up on `go test -v`?